networkx/algorithms/community/label_propagation.py

```diff
@@ -0,0 +1,120 @@
+    communities. Not implemented for directed graphs.
+    
+    Parameters
+    ----------
+    G : graph
+    An undirected NetworkX graph.      
+    
+    Returns
+    -------
+    communities : list
+    List of the communities (sets)
+    
+    Raises
+    ------
+    NetworkXNotImplemented
+       If the graph is directed
+  
+    References
+    ----------
+    .. [1] Cordasco, G., & Gargano, L. (2010, December). Community detection 
+       via semi-synchronous label propagation algorithms. In Business 
+       Applications of Social Network Analysis (BASNA), 2010 IEEE International 
+       Workshop on (pp. 1-8). IEEE.
+    """
+    coloring = _color_network(G)
+    # Create a unique label for each node in the graph
+    labeling = {v: k for k, v in enumerate(G)}
+    while not _labeling_complete(labeling, G):
+        # Update the labels of every node.
+        for color, nodes in coloring.items():
+            for n in nodes:
+                _update_label(n , labeling, G)
+        
+    for label in set(labeling.values()):
+        yield set((x for x in labeling if labeling[x] == label))
+        
+def _color_network(G):
+    """Colors the network so that neighboring nodes all have distinct colors.
+       Returns a dict of set of nodes and also a lookup of nodes to colors, in 
+       a tuple in that order.
+    """
+    coloring = dict() # color => set(node)
+    lookup = dict() # node => color
+    finally_colored_nodes = set()
+    
+    for n in G:
+        if n not in lookup:
+            color = 0                 
+        else:
+            color = lookup[n]
+        lookup[n] = color
+        finally_colored_nodes.add(n)
+        if color not in coloring: 
+            coloring[color] = set()
+        coloring[color].add(n)
+        for q in G.neighbors(n):
+            if q not in finally_colored_nodes: 
+                lookup[q] = lookup[n] + 1
+    
+    return coloring
+
+def _labeling_complete(labeling, G):
+    """Determines whether or not LPA is done. It is complete when all nodes 
+       have a label that is in the set of highest frequency labels amongst its
+       neighbors.
+
+       Nodes with no neighbors are themselves a community and are therefore
+       labeled.
+    """    
+    return all(labeling[v] in 
+                    _select_labels_of_highest_frequency(v, labeling, G) 
+                        for v in G if len(G[v]) > 0)
+
+def _select_labels_of_highest_frequency(node, labeling, G):
+    """Finds all labels of maximum frequency. Specified freqs must be a mapping
+       from label to frequency of that label.          
+
+       Returns a set.
+    """   
+    if not G[node]: 
+        # Nodes with no neighbors are themselves a community and are labeled
+        # accordingly, hence the immediate if statement.
+        return {labeling[node]}
+    else:
+        # Compute the frequencies of each node label in the graph    
+        freqs = dict(Counter(labeling[q] for q in G[node]))
```

Comment:
A `Counter` is a dictionary, you don't need to wrap it in `dict`.

Also, the comment is not quite correct, it is computing frequencies of all neighbors of `node`, not each node in the graph.